### PR TITLE
SciCrunch abi plot test

### PR DESCRIPTION
**Pull request by alan-wu**
# Description

Add a test against abi-plot objects from returned SciCrunch query and make sure the paths provided are valid.
This new test will look for dataset 141, iterate through the abi-plot objects and make sure the file path is valid if it is specified by the isDescribedBy field.

## Type of change

Delete those that don't apply.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tests have been ran and tested against both SciCrunch dev and pr endpoints.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/test_scicrunch.py [hunk 1]
```diff
@@ -115,6 +115,34 @@ def test_response_version(client):
...
+        if json_data['result'][0]['version'] == '1.1.4':
+            assert len(json_data['result'][0]['abi-plot']) == 5
+            identifier = json_data['result'][0]["dataset_identifier"]
+            version = json_data['result'][0]["dataset_version"]
+            assert identifier == "141"
+            assert version == "3"
+            #Construct the file path prefix, it should be /s3-resource/141/3/files
+            path_prefix = '/'.join(('', 's3-resource', identifier, version, 'files'))
+            for plot in json_data['result'][0]['abi-plot']:
+                if plot['datacite']['isDescribedBy']['path'] != 'derivative//':
```
Comment (hsorby) [inline tests/test_scicrunch.py:137]: This concerns me.  I think we may need to have a chat to K-Core about why we are having to do this.

File: tests/test_scicrunch.py [hunk 2]
```diff
@@ -115,6 +115,34 @@ def test_response_version(client):
...
+                    path = '/'.join((path_prefix, plot['datacite']['isDescribedBy']['path']))
+                    #Check if the file can be downloaded using the /s3-resource/{path} route
+                    r2 = client.get(path)
```
Comment (hsorby) [inline tests/test_scicrunch.py:140]: Would it be better to use the `exists` endpoint here?

**Review by hsorby — COMMENTED** [review]
One real question on this and one note to self to take further action.

File: tests/test_scicrunch.py (hunk 2 above)
Comment (alan-wu) [reply tests/test_scicrunch.py:140]: Sure, I did not notice this route already exists.

File: tests/test_scicrunch.py (hunk 1 above)
Comment (alan-wu) [reply tests/test_scicrunch.py:137]: Agreed, it should be an empty string if it is not specified.
I do not think this should stop the pull request from merging in once I address the following conversation
.

Comment (hsorby) [reply tests/test_scicrunch.py:137]: No indeed not, more of a note to self.

File: tests/test_scicrunch.py (hunk 2 above)
Comment (alan-wu) [reply tests/test_scicrunch.py:140]: Done, thanks.

**Review by hsorby — APPROVED** [review]

**Pull request merged**